packages/aws-cdk-lib/aws-opensearchservice/lib/domain.ts

```diff
@@ -1871,6 +1920,20 @@ export class Domain extends DomainBase implements IDomain, ec2.IConnectable {
       this.validateSamlAuthenticationOptions(props.fineGrainedAccessControl?.samlAuthenticationOptions);
     }
 
+    if (props.capacity?.nodeOptions) {
+      // Validate coordinator node configuration
+      const coordinatorConfig = props.capacity.nodeOptions.find(opt => opt.nodeType === 'coordinator')?.nodeConfig;
+      if (coordinatorConfig?.enabled) {
```

Comment:
Shouldn't we validate regardless of whether the node is enabled or not?

Reply:
So the structure looks like 
NodeOptions: [
{NodeType:"coordinator", "NodeConfig": {"Type": "m5.large.search", "Count": 1, "Enabled": true}},
...
]

So each node has their own config and we check their enablement only by enabled function call.